app/uk/gov/hmrc/homeofficesettledstatusstubs/services/StubDataService.scala

```diff
@@ -19,92 +19,83 @@ package uk.gov.hmrc.homeofficesettledstatusstubs.services
 import play.api.mvc._
 import uk.gov.hmrc.homeofficesettledstatusstubs.models._
 import uk.gov.hmrc.homeofficesettledstatusstubs.stubdata.StubData
+
 import javax.inject.{Inject, Singleton}
 import play.api.libs.json.Json
+import uk.gov.hmrc.homeofficesettledstatusstubs.models.searches._
 import uk.gov.hmrc.play.bootstrap.backend.controller.BackendController
 
 @Singleton
 class StubDataService @Inject()(cc: ControllerComponents) extends BackendController(cc) {
 
-  def ninoSearch(
-    correlationId: String,
-    nino: String,
-    givenName: String,
-    familyName: String,
-    dateOfBirth: String): Result =
-    resultFor(correlationId, nino) match {
+  private def search[S <: Searchable](searchable: S)(firstCheck: S => Option[StatusCheckResult])(
```

Comment:
Really like how we've abstracted the logic in this service.

Reply:
I think we could do something similar to this in the frontend when we introduce MRZ, where a NinoSearch and MrxSearch object are passed round generically, it will also help to determine things like where we redirect to on change/search again and which template/components to use to present data